blade/procData/views.py

```diff
@@ -3,49 +3,43 @@
 from procData.forms import uploadCSV
 from csv import DictReader
 
+
 # Create your views here.
 def breachStructure_list(request):
-	breachStructures = breachStructure.objects.all()
-
+    breachStructures = breachStructure.objects.all()
 
-	return render(request,
-		'breachStructure/list.html',
-		{'breachStructures':breachStructures})
+    return render(
+        request, "breachStructure/list.html", {"breachStructures": breachStructures}
+    )
 
 
 def breachStructure_detail(request, email):
-	breachStruct = get_object_or_404(breachStructure, email=email)
+    breachStruct = get_object_or_404(breachStructure, email=email)
 
+    return render(
+        request, "breachStructure/detail.html", {"breachStructures": breachStruct}
+    )
 
-	return render(request,
-		'breachStructure/detail.html',
-		{'breachStructures':breachStruct})
 
-#def query- username
 def upload_csv(request):
-	if request.method == "POST":
-		form = uploadCSV(request.POST, request.FILES)
-		if form.is_valid():
-			csv_file = request.FILES['csv_file']
-
-			reader = DictReader(csv_file.read().decode('utf-8').splitlines())
-			for row in reader:
-				print(row)
-				#dbQueriedUsers.objects.create(user=row[0])
+    if request.method == "POST":
+        form = uploadCSV(request.POST, request.FILES)
+        if form.is_valid():
+            csv_file = request.FILES["csv_file"]
 
-			return redirect('success_url')
+            reader = DictReader(csv_file.read().decode("utf-8").splitlines())
+            for row in reader:
+                user, created = dbQueriedUsers.objects.get_or_create(user=row[0])
```

Comment:
@abaddonpuff this way no duplicate entries can get into the table, but maybe this is not what you want actually, because there could be multiple csv files for different batches but with the same names. So feel free to change this back to just `create`, it was just to show you how you could do this.